crates/codegen/syntax_templates/src/typescript/cst_types.rs

```diff
@@ -75,59 +70,40 @@ impl TokenNode {
 
     #[napi(getter)]
     pub fn kind(&self) -> TokenKind {
-        match self.0.as_ref() {
-            Node::Token { kind, .. } => *kind,
-            _ => unreachable!(),
-        }
-    }
-
-    #[napi(getter, ts_return_type = "[ start: number, end: number ]")]
-    pub fn byte_range(&self) -> [u32; 2] {
-        let range = self.0.range();
-        [range.start.byte as u32, range.end.byte as u32]
-    }
-
-    #[napi(getter, ts_return_type = "[ start: number, end: number ]")]
-    pub fn char_range(&self) -> [u32; 2] {
-        let range = self.0.range();
-        [range.start.char as u32, range.end.char as u32]
+        self.0.kind
     }
 
-    #[napi(getter, ts_return_type = "[ start: number, end: number ]")]
-    pub fn byte_range_including_trivia(&self) -> [u32; 2] {
-        let range = self.0.range_including_trivia();
-        [range.start.byte as u32, range.end.byte as u32]
+    #[napi(getter)]
+    pub fn utf8_text_len(&self) -> u32 {
+        let text = &self.0.text;
+        text.len() as u32
     }
 
-    #[napi(getter, ts_return_type = "[ start: number, end: number ]")]
-    pub fn char_range_including_trivia(&self) -> [u32; 2] {
-        let range = self.0.range_including_trivia();
-        [range.start.char as u32, range.end.char as u32]
+    #[napi(getter)]
+    pub fn utf16_text_len(&self) -> u32 {
+        let text = &self.0.text;
+        text.encode_utf16().count() as u32
     }
 
-    #[napi(getter, ts_return_type = "(RuleNode | TokenNode)[]")]
-    pub fn trivia(&self, env: Env) -> Vec<JsObject> {
-        match self.0.as_ref() {
-            Node::Token { trivia, .. } => {
-                trivia.iter().map(|trivium| trivium.to_js(&env)).collect()
-            }
-            _ => unreachable!(),
-        }
+    #[napi(getter)]
+    pub fn char_text_len(&self) -> u32 {
+        let text = &self.0.text;
+        text.chars().count() as u32
     }
 }
 
 pub trait RcNodeExtensions {
     fn to_js(&self, env: &Env) -> JsObject;
 }
 
-impl RcNodeExtensions for Rc<Node> {
+impl RcNodeExtensions for Node {
     fn to_js(&self, env: &Env) -> JsObject {
```

Comment:
Is this meant to be exposed to te public API? does it need `#[napi]`?
If so, I suggest adding a test to get the object back and assert its JSON.

Reply:
no, it's just an internal conversion routine to wrap nodes into their TS objects